src/extensionApi.ts

```diff
@@ -149,10 +149,12 @@ export class CommandHandler {
     public async stopServer(forced: boolean, context?: ServerStateNode): Promise<Protocol.Status> {
         if (context === undefined) {
             const rsp = await this.selectRSP('Select RSP provider you want to retrieve servers');
-            if (!rsp || !rsp.id) return null;
+            if (!rsp || !rsp.id) 
+                return Promise.reject("No RSP chosen");
```

Comment:
In this case we don't need to show any error message because the user clicked 'esc' instead of choosing a rsp provider. So we just need to stop the workflow.